x/bridge/observer/bitcoin/bitcoin.go

```diff
@@ -16,170 +17,170 @@ import (
+)
 
-type Observer struct {
+type ChainClient struct {
 	logger             log.Logger
 	btcRpc             *rpcclient.Client
 	vaultAddr          string
-	currentHeight      uint64
-	globalTxInChan     chan TxIn
 	stopChan           chan struct{}
+	outboundChan       chan observer.Transfer
 	observeSleepPeriod time.Duration
+	lastObservedHeight uint64
 }
 
-// NewObserver returns new instance of `Observer` with BTC RPC client
-func NewObserver(
+// NewChainClient returns new instance of `Bitcoin`
+func NewChainClient(
 	logger log.Logger,
 	btcRpc *rpcclient.Client,
 	vaultAddr string,
-	initialHeight uint64,
 	observeSleepPeriod time.Duration,
-) (Observer, error) {
+	lastObservedHeight uint64,
+) (*ChainClient, error) {
 	if len(vaultAddr) == 0 {
-		return Observer{}, errorsmod.Wrapf(ErrInvalidCfg, "Invalid vaultAddr")
+		return nil, errorsmod.Wrapf(ErrInvalidCfg, "Invalid BTC vault address")
 	}
 
-	return Observer{
-		logger:             logger.With("module", ModuleNameObserver),
+	return &ChainClient{
+		logger:             logger.With("module", ModuleName),
 		btcRpc:             btcRpc,
 		vaultAddr:          vaultAddr,
-		currentHeight:      initialHeight,
-		globalTxInChan:     make(chan TxIn),
 		stopChan:           make(chan struct{}),
+		outboundChan:       make(chan observer.Transfer),
 		observeSleepPeriod: observeSleepPeriod,
+		lastObservedHeight: lastObservedHeight,
 	}, nil
 }
 
-// Start starts observing BTC blocks for incoming Txs to the given address
-func (o *Observer) Start() {
-	go o.observeBlocks()
+// Start starts observing Bitcoin blocks for outbound transfers
+func (b *ChainClient) Start(context.Context) error {
+	go b.observeBlocks()
+
+	return nil
+}
+
+// Stop stops observing Bitcoin blocks and shutdowns RPC client
+func (b *ChainClient) Stop(context.Context) error {
+	close(b.stopChan)
+	b.btcRpc.Shutdown()
+	b.btcRpc.WaitForShutdown()
+	return nil
+}
+
+// ListenOutboundTransfer returns receive-only channel with outbound transfer items
+func (b *ChainClient) ListenOutboundTransfer() <-chan observer.Transfer {
+	return b.outboundChan
+}
+
+// SignalInboundTransfer sends `InboundTransfer` to Bitcoin
+func (b *ChainClient) SignalInboundTransfer(ctx context.Context, in observer.Transfer) error {
+	return fmt.Errorf("Not implemented")
 }
 
-// Stop stops observation loop and RPC client
-func (o *Observer) Stop() {
-	close(o.stopChan)
-	o.btcRpc.Shutdown()
-	o.btcRpc.WaitForShutdown()
+// Returns current height of the Bitcoin chain
+func (b *ChainClient) Height() (uint64, error) {
+	height, err := b.btcRpc.GetBlockCount()
+	if err != nil {
+		return 0, errorsmod.Wrapf(ErrRpcClient, "Failed to get current height %s", err.Error())
+	}
+	return uint64(height), nil
 }
 
-// TxIns returns receive-only part of observed Txs channel
-func (o *Observer) TxIns() <-chan TxIn {
-	return o.globalTxInChan
+// Returns number of required tx confirmations
+func (b *ChainClient) ConfirmationsRequired() (uint64, error) {
+	return 0, nil
```

Comment:
I left a similar comment on the other PR but I want to make sure this is going to not return 0 later